code/server/server.cpp

```diff
@@ -0,0 +1,178 @@
+#include <iostream>
+#include <yarp/os/all.h>
+
+using namespace std;
+using namespace yarp::os;
+
+#define FIFO_SIZE 10
+
+volatile int intFIFOtmp[FIFO_SIZE];
+volatile int intFIFOnum;
+
+class FIFO{
+private:
+	int intHead;
+	int intQueLen;
+	int intFIFO[FIFO_SIZE];
+public:
```

Comment:
I didn't know that C++ has such a useful vector!
I fixed to use this vector in my code.